Major L5 updates

Tiniifan opened the pull request:
Hi I tried to improve Metanoia for some L5 files, the code isn't perfect but I think it's a good start:
- remove the restriction from RES to view iego, yokai 3D model
- support XRES from iego
- add mtn3 support for motion
- add mtninf support
- use huffman decompression from kuriimu2 instead of metanoia huffman decompression
- use rle decompression from kuriimu2 instead of metanoia rle decompression
- can import animations from xpck (.xc)
- fix rgb8 decoder

Tiniifan committed:
Major L5 updates

- target end of line to windows format
- remove the restriction from RES to view iego, yokai 3D model
- support XRES from iego
- add mtn3 support for motion
- add mtninf support
- use huffman decompression from kuriimu2 instead of metanoia huffman decompression
- use rle decompression from kuriimu2 instead of metanoia rle decompression
- can import animations from xpck (.xc)

Tiniifan committed:
fixed mtn3 animoffset

Tiniifan committed:
improve animation reader/exporter

- fix mtn3 loader
- can view splitted animation name from mtn2
- can export multiple animation

Tiniifan committed:
fix mtn frame reader

Metanoia had a problem retrieving frames and could not read a model with more than 255 frames

Tiniifan committed:
Add zoom feature with mouse wheel

in1tiate commented:
Fixes broken animation loading from Professor Layton VS. Phoenix Wright: Ace Attorney


https://github.com/Ploaj/Metanoia/assets/32779090/8a7bc245-1792-4c2f-b9de-6e1340d493c6

Tiniifan commented:
According to this [issue](https://github.com/Ploaj/Metanoia/issues/9)

The problem was due to an error in reading the number of frames for a bone. 
Metanoia can't read an animation with more than 255 frames, 
because the frame number is set to 2 bytes (the hight byte starts at 0x20 instead of 0x00) and metanoia reads 1 byte.

Tiniifan committed:
Fix compression and texture problem

- Use Kuriimu LZ10 for better decompression
- Fix compression header type
- Fix problem with some XI (large texture)

smile-files commented:
Hi, sorry! I'm very new to messing around with game files, and being on GitHub generally, so apologies if this isn't the right place to ask these questions (and apologies if these questions are silly).

![image](https://github.com/user-attachments/assets/f8be667c-b9a4-48d6-bb47-3a431325e383)
Occasionally I get an error such as this when trying to open an .xc file as a model... 

![image](https://github.com/user-attachments/assets/235fd299-be53-4d15-a53a-a81847df1fa8)
![image](https://github.com/user-attachments/assets/62da1771-d860-4b1b-bee0-5fc1511dcb36)
Is there any way for me to troubleshoot such that I can open this model? If it is an issue with the texture (the error message cites "Unknown Texture Type 0") would messing with the model's texture file do anything? Which of these files under the dropdown would that be?

If it's of any use to you, I'm working with the files for Professor Layton and the Miracle Mask... And again, sorry if this is the wrong question to ask in the wrong place! It is hard to find people online talking about this program. Thank you so much!

Tiniifan commented:
Hi.

This isn’t the right place for this kind of request. Please avoid posting such questions in a pull request discussion, as this thread is specifically meant for reviewing and discussing the changes proposed in the PR.

For context, there are two versions of Metanoia:

* **The official version by Ploaj** — available here: [https://github.com/Ploaj/Metanoia](https://github.com/Ploaj/Metanoia)
  → If you’re experiencing issues with Ploaj’s version, please open an issue on that repository: [https://github.com/Ploaj/Metanoia/issues](https://github.com/Ploaj/Metanoia/issues)

* **My fork of Metanoia** — which includes some fixes and improvements. You can download it here: [https://github.com/Tiniifan/Metanoia/releases/tag/LatestBuild](https://github.com/Tiniifan/Metanoia/releases/tag/LatestBuild)
  → If your issue relates to my fork, please open an issue here: [https://github.com/Tiniifan/Metanoia/issues](https://github.com/Tiniifan/Metanoia/issues)

In short, please use the relevant issue place for support requests instead of a pull request discussion, as it’s important for us to know exactly which version you’re using and keep discussions organized.
